storage/response_cache.go

```diff
@@ -72,26 +72,32 @@ func (rc *ResponseCache) ClearData(e engine.Engine) error {
 	return err
 }
 
-// GetResponse looks up a response matching the specified cmdID and
-// returns true if found. The response is deserialized into the
-// supplied reply parameter. If no response is found, returns
-// false. If a command is pending already for the cmdID, then this
-// method will block until the the command is completed or the
-// response cache is cleared.
-func (rc *ResponseCache) GetResponse(e engine.Engine, cmdID proto.ClientCmdID) (proto.Response, error) {
+// GetResponse looks up a response matching the specified cmdID. If the
+// response is found, it is returned along with its associated error.
+// If the response is not found, nil is returned for both the response
+// and its error. In all cases, the third return value is a
+// *replicaCorruptionError that is non-nil when the response cache is
+// unable to read from storage.
+func (rc *ResponseCache) GetResponse(e engine.Engine, cmdID proto.ClientCmdID) (proto.Response, error, error) {
```

Comment:
Does errcheck support multiple error returns from one function? Regardless, I'm not concerned at all about the loss of tool support implied by ResponseWithError. We've made huge gains in tool support by limiting nonstandard error patterns to a few places; we can trust ourselves to do the right thing with the return value from ResponseCache.GetResponse whether it's an error or a ResponseWithError.

Anyway, I don't feel strongly about it and silencing golint is probably better than letting it drive the design of our code. I just wish there was a less-ugly way to suppress this warning.